parsel/selector.py

```diff
@@ -175,8 +175,11 @@ def xpath(self, query):
             return self.selectorlist_cls([])
 
         try:
-            result = xpathev(query, namespaces=self.namespaces,
-                             smart_strings=self._lxml_smart_strings)
+            nsp = dict(self.namespaces)
+            nsp.update(kwargs.pop('namespaces', {}))
```

Comment:
> why not adding namespaces as a keyword function argument with default value?

Indeed.

For the 2nd point, I'd go for 1. `.xpath('//input[@value=$number]/@name', parsel_namespaces={'foo': 'bar'}, number=1)`.